### PR TITLE
CCv0: Merge main into CCv0 branch

### PR DESCRIPTION
**Pull request by GeorginaKin**
Merge remote-tracking branch 'upstream/main' into CCv0

Fixes: #6428
Signed-off-by: Georgina Kinge <georgina.kinge@ibm.com>

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ assets:
       url: "https://github.com/cloud-hypervisor/cloud-hypervisor"
       uscan-url: >-
         https://github.com/cloud-hypervisor/cloud-hypervisor/tags.*/v?(\d\S+)\.tar\.gz
-      version: "v29.0"
+      version: "v30.0"
```
Comment (stevenhorsman) [inline versions.yaml:78]: Hi @GeorginaKin - can we revert this change as I don't think CLH should be updated in `CC`

Comment (GeorginaKin) [reply versions.yaml]: Ah yeah apologies! Totally forgot with all the conflicts going on- thank you!!

**Comment by fidencio** [comment]
@GeorginaKin, first of all, thanks a lot for already reverting https://github.com/kata-containers/kata-containers/pull/6440/commits/262daaa2eff4ab725df498ea841f2893e4cc94e3.

We also need to revert https://github.com/kata-containers/kata-containers/pull/6440/commits/3ac6f29e95446e318b682d96bcba8a392ff5f392.

File: tools/packaging/static-build/qemu/build-base-qemu.sh [hunk 2]
```diff
@@ -38,16 +38,15 @@ CACHE_TIMEOUT=$(date +"%Y-%m-%d")
 [ -n "${build_suffix}" ] && HYPERVISOR_NAME="kata-qemu-${build_suffix}" || HYPERVISOR_NAME="kata-qemu"
 [ -n "${build_suffix}" ] && PKGVERSION="kata-static-${build_suffix}" || PKGVERSION="kata-static"
 
-container_image="${QEMU_CONTAINER_BUILDER:-$(get_qemu_image_name)}"
+container_image="${QEMU_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:qemu-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/qemu/build-base-qemu.sh:41]: I think this needs reverting to the old approach to pick up `CC_BUILDER_REGISTRY` from lib.sh rather than `BUILDER_REGISTRY`
```suggestion
container_image="${QEMU_CONTAINER_BUILDER:-$(get_qemu_image_name)}"
```

File: tools/packaging/static-build/ovmf/build.sh [hunk 3]
```diff
@@ -15,7 +15,7 @@ source "${script_dir}/../../scripts/lib.sh"
 
 DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
-container_image="${OVMF_CONTAINER_BUILDER:-$(get_ovmf_image_name)}"
+container_image="${OVMF_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:ovmf-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/ovmf/build.sh:18]: I think this needs reverting to the old approach to pick up `CC_BUILDER_REGISTRY` from lib.sh rather than ```suggestion
container_image="${OVMF_CONTAINER_BUILDER:-$(get_ovmf_image_name)}"
```

File: tools/packaging/static-build/kernel/build.sh [hunk 4]
```diff
@@ -16,12 +16,12 @@ readonly kernel_builder="${repo_root_dir}/tools/packaging/kernel/build-kernel.sh
 
 DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
-container_image="${KERNEL_CONTAINER_BUILDER:-$(get_kernel_image_name)}"
+container_image="${KERNEL_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:kernel-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/kernel/build.sh:19]: Same here:
```suggestion
container_image="${KERNEL_CONTAINER_BUILDER:-$(get_kernel_image_name)}"
```

File: tools/packaging/static-build/shim-v2/build.sh [hunk 5]
```diff
@@ -20,17 +22,14 @@ RUST_VERSION=${RUST_VERSION}
 
 DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
-container_image="${SHIM_V2_CONTAINER_BUILDER:-$(get_shim_v2_image_name)}"
-
-EXTRA_OPTS="${EXTRA_OPTS:-""}"
-VMM_CONFIGS="qemu fc"
-REMOVE_VMM_CONFIGS="${REMOVE_VMM_CONFIGS:-""}"
+container_image="${SHIM_V2_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:shim-v2-go-${GO_VERSION}-rust-${RUST_VERSION}-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/shim-v2/build.sh:25]: Same here:
```suggestion
container_image="${SHIM_V2_CONTAINER_BUILDER:-$(get_shim_v2_image_name)}"
```

File: tools/packaging/static-build/td-shim/build.sh [hunk 6]
```diff
@@ -29,15 +29,14 @@ package_output_dir="${package_output_dir:-}"
...
-container_image="${TDSHIM_CONTAINER_BUILDER:-$(get_td_shim_image_name)}"
-
-sudo docker pull ${container_image} || \
-	(sudo docker build \
-		--build-arg RUST_TOOLCHAIN="${tdshim_toolchain}" \
-		-t "${container_image}" \
-		"${script_dir}" && \
-	 # No-op unless PUSH_TO_REGISTRY is exported as "yes"
-	 push_to_registry "${container_image}")
+container_image="${TDSHIM_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:td-shim-${tdshim_toolchain}-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/td-shim/build.sh:32]: Same here
```suggestion
container_image="${TDSHIM_CONTAINER_BUILDER:-$(get_td_shim_image_name)}"
```

File: tools/packaging/static-build/virtiofsd/build.sh [hunk 7]
```diff
@@ -48,7 +48,7 @@ case ${ARCH} in
 		;;
 esac
 
-container_image="${VIRTIOFSD_CONTAINER_BUILDER:-$(get_virtiofsd_image_name)}"
+container_image="${VIRTIOFSD_CONTAINER_BUILDER:-${BUILDER_REGISTRY}:virtiofsd-${virtiofsd_toolchain}-${libc}-$(get_last_modification ${script_dir})-$(uname -m)}"
```
Comment (stevenhorsman) [inline tools/packaging/static-build/virtiofsd/build.sh:51]: Same here:
```suggestion
container_image="${VIRTIOFSD_CONTAINER_BUILDER:-$(get_virtiofsd_image_name)}"
```

**Review by stevenhorsman — CHANGES_REQUESTED** [review]
We want to udpate the container image names to use the lib.sh utils methods

**Comment by stevenhorsman** [comment]
@GeorginaKin - FYI the static check is failing on the kernel version check.
The kernel has been changed by SEV in both main and CCv0 , but both bumped the kernel config version to 101, so the merge didn't update the version for caching even though there is a kernel diff, so the static check is failing.
To resolve the problem now we can bump the version in `tools/packaging/kernel/kata_config_version`, but I'm trying to work out if there would be an knock on issues for this later on

**Comment by GeorginaKin** [comment]
```
git diff --stat upstream/main -- . ':!*/vendor/*'
 .github/workflows/cc-payload-after-push-amd64.yaml                                              |  159 ++
 .github/workflows/cc-payload-after-push-s390x.yaml                                              |  153 ++
 .github/workflows/cc-payload-after-push.yaml                                                    |   39 +
 .github/workflows/cc-payload-amd64.yaml                                                         |  142 ++
 .github/workflows/cc-payload-s390x.yaml                                                         |  134 ++
 .github/workflows/cc-payload.yaml                                                               |   39 +
 .github/workflows/commit-message-check.yaml                                                     |    4 +-
 .github/workflows/deploy-ccv0-demo.yaml                                                         |  124 ++
 .github/workflows/static-checks.yaml                                                            |    4 +-
 ci/install_libseccomp.sh                                                                        |    3 +-
 docs/how-to/README.md                                                                           |    3 +
 docs/how-to/ccv0.sh                                                                             |  640 ++++++++
 docs/how-to/data/confidential-agent-config.toml.in                                              |   45 +
 docs/how-to/how-to-build-and-test-ccv0.md                                                       |  475 ++++++
 .../how-to-generate-a-kata-containers-payload-for-the-confidential-containers-operator.md       |   44 +
 docs/how-to/how-to-setup-swap-devices-in-guest-kernel.md                                        |    2 +
 docs/how-to/how-to-use-virtio-fs-nydus-with-kata.md                                             |    1 +
 docs/how-to/how-to-use-virtio-mem-with-kata.md                                                  |    2 +
 src/agent/Cargo.lock                                                                            | 4077 ++++++++++++++++++++++++++++++++++++++++------
 src/agent/Cargo.toml                                                                            |   14 +-
 src/agent/rustjail/src/cgroups/fs/mod.rs                                                        |   19 +-
 src/agent/rustjail/src/lib.rs                                                                   |    8 +-
 src/agent/src/config.rs                                                                         |  220 ++-
 src/agent/src/device.rs                                                                         |   18 +-
 src/agent/src/image_rpc.rs                                                                      |  337 ++++
 src/agent/src/main.rs                                                                           |    1 +
 src/agent/src/rpc.rs                                                                            |  323 +++-
 src/agent/src/sandbox.rs                                                                        |    2 +
 src/agent/src/signal.rs                                                                         |    2 +-
 src/dragonball/src/api/v1/instance_info.rs                                                      |   20 +-
 src/dragonball/src/api/v1/mod.rs                                                                |    2 +-
 src/dragonball/src/error.rs                                                                     |    4 +
 src/dragonball/src/kvm_context.rs                                                               |   47 +-
 src/dragonball/src/vm/mod.rs                                                                    |   30 +-
 src/libs/oci/src/lib.rs                                                                         |   10 +-
 src/libs/protocols/.gitignore                                                                   |    3 +
 src/libs/protocols/build.rs                                                                     |   21 +-
 src/libs/protocols/protos/image.proto                                                           |   31 +
 src/libs/protocols/src/lib.rs                                                                   |    4 +
 src/libs/protocols/src/trans.rs                                                                 |    4 +-
 src/runtime/Makefile                                                                            |  122 ++
 src/runtime/arch/amd64-options.mk                                                               |    2 +
 src/runtime/cmd/kata-runtime/kata-check_amd64.go                                                |    2 +
 src/runtime/cmd/kata-runtime/kata-check_arm64.go                                                |    3 +
 src/runtime/cmd/kata-runtime/kata-check_ppc64le.go                                              |    3 +
 src/runtime/cmd/kata-runtime/kata-check_s390x.go                                                |    3 +
 src/runtime/config/configuration-clh-tdx.toml.in                                                |  443 +++++
 src/runtime/config/configuration-clh.toml.in                                                    |    2 +-
 src/runtime/config/configuration-qemu-sev.toml.in                                               |  675 ++++++++
 src/runtime/config/configuration-qemu-tdx.toml.in                                               |  651 ++++++++
 src/runtime/config/configuration-qemu.toml.in                                                   |   33 +-
 src/runtime/go.mod                                                                              |    3 +-
 src/runtime/go.sum                                                                              |  282 +---
 src/runtime/hack/update-generated-hypervisor-proto.sh                                           |   13 +
 src/runtime/pkg/README.md                                                                       |    1 +
 src/runtime/pkg/containerd-shim-v2/create.go                                                    |    5 +
 src/runtime/pkg/containerd-shim-v2/image.go                                                     |   84 +
 src/runtime/pkg/govmm/qemu/qemu.go                                                              |   45 +-
 src/runtime/pkg/govmm/qemu/qmp.go                                                               |   66 +
 src/runtime/pkg/katautils/config-settings.go.in                                                 |    8 +
 src/runtime/pkg/katautils/config.go                                                             |  288 ++--
 src/runtime/pkg/katautils/config_test.go                                                        |    2 +
 src/runtime/pkg/oci/utils.go                                                                    |    5 +
 src/runtime/pkg/oci/utils_test.go                                                               |    2 +-
 src/runtime/pkg/sev/README.md                                                                   |   14 +
 src/runtime/pkg/sev/ovmf.go                                                                     |  101 ++
 src/runtime/pkg/sev/sev.go                                                                      |  217 +++
 src/runtime/pkg/sev/sev_test.go                                                                 |   54 +
 src/runtime/pkg/sev/testdata/README.md                                                          |    9 +
 src/runtime/pkg/sev/testdata/ovmf_suffix.bin                                                    |  Bin 0 -> 4096 bytes
 src/runtime/pkg/sev/vcpu_sigs.go                                                                |   76 +
 src/runtime/pkg/sev/vcpu_sigs_test.go                                                           |   21 +
 src/runtime/pkg/sev/vmsa.go                                                                     |  172 ++
 src/runtime/protocols/hypervisor/hypervisor.pb.go                                               | 1848 +++++++++++++++++++++
 src/runtime/protocols/hypervisor/hypervisor.proto                                               |   46 +
 src/runtime/protocols/simple-kbs/keybroker.pb.go                                                |  573 +++++++
 src/runtime/protocols/simple-kbs/keybroker.proto                                                |   68 +
 src/runtime/protocols/simple-kbs/keybroker_grpc.pb.go                                           |  141 ++
 src/runtime/virtcontainers/acrn.go                                                              |    9 +
 src/runtime/virtcontainers/agent.go                                                             |    3 +
 src/runtime/virtcontainers/clh.go                                                               |   15 +-
 src/runtime/virtcontainers/container.go                                                         |    5 +
 src/runtime/virtcontainers/fc.go                                                                |    4 +
 src/runtime/virtcontainers/fs_share_linux.go                                                    |   57 +-
 src/runtime/virtcontainers/hypervisor.go                                                        |  444 ++---
 src/runtime/virtcontainers/hypervisor_config_darwin.go                                          |    4 +
 src/runtime/virtcontainers/hypervisor_config_linux.go                                           |    4 +
 src/runtime/virtcontainers/hypervisor_linux.go                                                  |    2 +
 src/runtime/virtcontainers/image/image.go                                                       |   24 +
 src/runtime/virtcontainers/interfaces.go                                                        |    3 +
 src/runtime/virtcontainers/kata_agent.go                                                        |  155 +-
 src/runtime/virtcontainers/mock_agent.go                                                        |    5 +
 src/runtime/virtcontainers/mock_hypervisor.go                                                   |    4 +
 src/runtime/virtcontainers/monitor.go                                                           |   11 +-
 src/runtime/virtcontainers/mount.go                                                             |   24 +-
 src/runtime/virtcontainers/pkg/agent/protocols/client/client.go                                 |   35 +
 src/runtime/virtcontainers/pkg/agent/protocols/grpc/image.pb.go                                 |  613 +++++++
 src/runtime/virtcontainers/pkg/cloud-hypervisor/client/api/openapi.yaml                         |    3 -
 src/runtime/virtcontainers/pkg/cloud-hypervisor/client/docs/CpuAffinity.md                      |   16 +-
 src/runtime/virtcontainers/pkg/cloud-hypervisor/client/model_cpu_affinity.go                    |   64 +-
 src/runtime/virtcontainers/pkg/cloud-hypervisor/cloud-hypervisor.yaml                           |    3 -
 src/runtime/virtcontainers/pkg/mock/mock.go                                                     |    4 +
 src/runtime/virtcontainers/pkg/vcmock/sandbox.go                                                |    5 +
 src/runtime/virtcontainers/qemu.go                                                              |  120 +-
 src/runtime/virtcontainers/qemu_amd64.go                                                        |  242 ++-
 src/runtime/virtcontainers/qemu_amd64_test.go                                                   |   24 +-
 src/runtime/virtcontainers/qemu_arch_base.go                                                    |   56 +-
 src/runtime/virtcontainers/remote.go                                                            |  298 ++++
 src/runtime/virtcontainers/sandbox.go                                                           |   74 +-
 src/runtime/virtcontainers/types/sandbox.go                                                     |   12 +
 src/tools/agent-ctl/src/client.rs                                                               |  117 +-
 src/tools/agent-ctl/src/utils.rs                                                                |    2 +-
 tools/osbuilder/image-builder/Dockerfile                                                        |    1 +
 tools/osbuilder/image-builder/image_builder.sh                                                  |   22 +-
 tools/osbuilder/rootfs-builder/README.md                                                        |    2 +-
 tools/osbuilder/rootfs-builder/agent-config.toml.in                                             |   44 +
 tools/osbuilder/rootfs-builder/alpine/Dockerfile.in                                             |    4 +
 tools/osbuilder/rootfs-builder/clearlinux/Dockerfile.in                                         |    5 +
 tools/osbuilder/rootfs-builder/init_trusted_storage.sh                                          |  130 ++
 tools/osbuilder/rootfs-builder/rootfs.sh                                                        |   63 +
 tools/osbuilder/rootfs-builder/ubuntu/Dockerfile.in                                             |   26 +-
 tools/osbuilder/rootfs-builder/ubuntu/config.sh                                                 |   21 +-
 tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh                                             |   28 +-
 tools/osbuilder/scripts/lib.sh                                                                  |    9 +
 tools/packaging/guest-image/build_image.sh                                                      |   60 +-
 tools/packaging/kata-deploy-cc/Dockerfile                                                       |   31 +
 tools/packaging/kata-deploy-cc/scripts/kata-deploy.sh                                           |  288 ++++
 tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml                                 |    2 +-
 tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml                                   |    4 +-
 tools/packaging/kata-deploy/local-build/Makefile                                                |   70 +
 tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile                                  |    1 +
 tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh                       |    7 +-
 tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh                                 |  494 +++++-
 tools/packaging/kata-deploy/local-build/kata-deploy-build-and-upload-payload.sh                 |   23 +-
 tools/packaging/kata-deploy/scripts/kata-deploy.sh                                              |    0
 tools/packaging/kernel/build-kernel.sh                                                          |   37 +-
 tools/packaging/kernel/configs/fragments/common/confidential_containers/cryptsetup.conf         |   21 +
 tools/packaging/kernel/configs/fragments/common/confidential_containers/initramfs.conf          |    1 +
 tools/packaging/kernel/configs/fragments/x86_64/crypto.conf                                     |    3 +
 tools/packaging/kernel/kata_config_version                                                      |    2 +-
 tools/packaging/qemu/patches/7.1.x/no_patches.txt                                               |    0
 tools/packaging/qemu/patches/tag_patches/v7.1.0/no_patches.txt                                  |    0
 tools/packaging/scripts/configure-hypervisor.sh                                                 |    4 +-
 tools/packaging/scripts/lib.sh                                                                  |  115 +-
 tools/packaging/static-build/cache_components.sh                                                |  270 +++
 tools/packaging/static-build/initramfs/Dockerfile                                               |   68 +
 tools/packaging/static-build/initramfs/build-initramfs.sh                                       |   15 +
 tools/packaging/static-build/initramfs/build.sh                                                 |   47 +
 tools/packaging/static-build/initramfs/init.sh                                                  |   44 +
 tools/packaging/static-build/initramfs/initramfs.list                                           |   21 +
 tools/packaging/static-build/kernel/Dockerfile                                                  |    1 +
 tools/packaging/static-build/kernel/build.sh                                                    |    3 +-
 tools/packaging/static-build/ovmf/Dockerfile                                                    |   10 +-
 tools/packaging/static-build/ovmf/build-ovmf.sh                                                 |    7 +-
 tools/packaging/static-build/ovmf/build.sh                                                      |   18 +-
 tools/packaging/static-build/qemu/build-base-qemu.sh                                            |    2 +-
 tools/packaging/static-build/qemu/build-static-qemu-cc.sh                                       |   34 +
 tools/packaging/static-build/shim-v2/build.sh                                                   |   14 +-
 tools/packaging/static-build/td-shim/Dockerfile                                                 |    4 +-
 tools/packaging/static-build/td-shim/build-td-shim.sh                                           |    4 +-
 tools/packaging/static-build/td-shim/build.sh                                                   |    2 +-
 tools/packaging/static-build/virtiofsd/build.sh                                                 |    2 +-
 utils.mk                                                                                        |    4 +-
 versions.yaml                                                                                   |   59 +-
 164 files changed, 16306 insertions(+), 1595 deletions(-)
```

**Comment by fidencio** [comment]
http://10.112.240.208:8080/job/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/757/ is green but didn't report bacl.

File: tools/packaging/static-build/shim-v2/build.sh [hunk 8]
```diff
@@ -22,15 +24,12 @@ DESTDIR=${DESTDIR:-${PWD}}
 PREFIX=${PREFIX:-/opt/kata}
 container_image="${SHIM_V2_CONTAINER_BUILDER:-$(get_shim_v2_image_name)}"
 
-EXTRA_OPTS="${EXTRA_OPTS:-""}"
-VMM_CONFIGS="qemu fc"
-REMOVE_VMM_CONFIGS="${REMOVE_VMM_CONFIGS:-""}"
-
```
Comment (stevenhorsman) [inline tools/packaging/static-build/shim-v2/build.sh:28]: @GeorginaKin - we need all these lines still, so if you can add them back in, that would be great.

**Comment by stevenhorsman** [comment]
The static checks are failing/cancelled due to issues with the GHA runner:
```
[static-checks (sudo -E PATH="$PATH" make test)](https://github.com/kata-containers/kata-containers/actions/runs/4407205134/jobs/7743414807#step:1:38)
You are running out of disk space. The runner will stop working when the machine runs out of disk space. Free space left: 67 MB
```

**Review by stevenhorsman — APPROVED** [review]
LGTM now!

**Comment by fidencio** [comment]
```
11:35:38 # Mar 15 18:33:57 tdx-ci kata[2831532]: time="2023-03-15T18:33:57.589254848+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d/console.sock name=containerd-shim-v2 pid=2831532 sandbox=764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d source=virtcontainers subsystem=sandbox vmconsole="[ERROR] failed to ioctl get tdx report data."
11:35:38 # Mar 15 18:33:57 tdx-ci kata[2831532]: time="2023-03-15T18:33:57.589282341+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d/console.sock name=containerd-shim-v2 pid=2831532 sandbox=764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d source=virtcontainers subsystem=sandbox vmconsole="[ERROR] failed to get tdx report."
11:35:38 # Mar 15 18:33:57 tdx-ci kata[2831532]: time="2023-03-15T18:33:57.589309299+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d/console.sock name=containerd-shim-v2 pid=2831532 sandbox=764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d source=virtcontainers subsystem=sandbox vmconsole="[ERROR] failed to generate quote"
11:35:38 # Mar 15 18:33:57 tdx-ci kata[2831532]: time="2023-03-15T18:33:57.58934709+08:00" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d/console.sock name=containerd-shim-v2 pid=2831532 sandbox=764a26d44f8d5787f3cb6aa6d140ab83f7c8436e8016b01dc68f7da633859c4d source=virtcontainers subsystem=sandbox vmconsole="\x1b[0m\x1b[38;5;8m[\x1b[0m2023-03-15T10:33:56Z \x1b[0m\x1b[1m\x1b[31mERROR\x1b[0m attestation_agent::kbc_modules::eaa_kbc::rats_tls\x1b[0m\x1b[38;5;8m]\x1b[0m rats_tls_init() failed"
```

Failure on TDX is legit, but after a conversation with @arronwy we think it's better to not block this PR to get merged.
So, please, go ahead with this and its counterpart on the tests repo.
